pymc/sampling/mcmc.py

```diff
@@ -212,9 +212,81 @@ def all_continuous(vars):
         return True
 
 
+def try_fast_nuts(
+    draws: int = 1000,
+    initvals: Optional[Union[StartDict, Sequence[Optional[StartDict]]]] = None,
+    chains: Optional[int] = None,
+    cores: Optional[int] = None,
+    tune: int = 1000,
+    progressbar: bool = True,
+    model=None,
+    random_seed: RandomState = None,
+    discard_tuned_samples: bool = True,
+    **kwargs,
+):
+    target_accept = kwargs.get("nuts", {}).get("target_accept", 0.8)
+    # Try different fast samplers in order of their speed
+    try:
+        _log.info("Trying nutpie sampler...")
+        import nutpie
+
+        compiled_model = nutpie.compile_pymc_model(model)
+        idata = nutpie.sample(
+            compiled_model,
+            draws=draws,
+            tune=tune,
+            chains=chains,
+            seed=random_seed,
+            save_warmup=~discard_tuned_samples,
+        )
+        return idata
+    except ImportError:
+        _log.info("nutpie not found. Install it with mamba install -c conda-forge nutpie")
+    except Exception as e:
+        _log.info(e)
+    try:
+        _log.info("Trying NumPyro JAX sampler...")
+        import pymc.sampling.jax as pymc_jax
```

Comment:
Only the import should be in the try block, sampling exceptions should raise naturally, no?

Reply:
I wanted to hedge a little bit in case the user is e.g. running into numba ops that do not exist yet, we want to fail gracefully.

Reply:
What about just accepting the sampler as a string and if it fails the user can try the next one?

`nuts_sampler : ["nutpie", "numpyro", "blackjax", "pymc"] = "pymc"`

Reply:
yeah, that was my first implementation. Reconsidering, that might be better.

Reply:
What I don't like about this is that we can't easily switch it on as a default and have a good fallback.

Reply:
When we think the time is ready we can start announcing that the default will be nutpie and tell people to explicit set it to "pymc" if they don't want it to change.

I think we will be able to do the switch without to much pain that we don't need to worry about a runtime fallback.

On a separate note, it would be nice to include a sentence about reporting issues to us when attempting to use the nutpie sampler

Reply:
Good point, with conda-forge installation should be fine.

Reply:
I like the string argument, particularly since the `fit` function already uses a `method` argument to specify the algorithm.

Reply:
OK, implemented the change.